packages/components/src/navigator/navigator-switch.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * External dependencies
+ */
+import { AnimatePresence } from 'framer-motion';
+
+/**
+ * WordPress dependencies
+ */
+import { memo } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { Route, Switch } from './router';
+
+function NavigatorSwitch( { children } ) {
+	return (
+		<Route
+			render={ ( { location } ) => (
+				<AnimatePresence initial={ false } exitBeforeEnter>
+					<Switch key={ location.pathname } location={ location }>
+						{ children }
+					</Switch>
+				</AnimatePresence>
+			) }
+		/>
+	);
+}
+
+export default memo( NavigatorSwitch );
```

Comment:
Is it necessary to `memo` this component? Its only prop is `children`.

Reply:
Agreed - I don't see why we'd need a `memo` here at a first glance. Just moved it as it was part of the prototype PR.